### PR TITLE
All guides

### LINKED ISSUE #144: New all guides page design
New page design for the view that shows all the guides
reference_all guides table of contents

### PR DESCRIPTION
**Pull request by steven1046**
- What was fixed?  (Issue # or description of fix)

Update the all guides page to reflect the overall docs layout.
Extract the doc header css into a separate file.
Add a sticky footer to the all guides page when filtering guides so it stays at the bottom.
Removed static width for guide cards on specific media widths to make the layout smoother.

This pull request is a copy of #163 since a rebase was done on top of development, then the SHAs changed and Github will not let you reopen a pull request when you force push to a branch to override old SHAs.

Fixes #144
- Were the changes tested on
- [x] Firefox (Desktop)
- [x] Safari (Desktop)
- [x] Chrome (Desktop)
- [ ] Internet Explorer (Desktop)
- [ ] iOS (Mobile)
- [ ] Android (Mobile)
- Running validation tools
    - [x] https://validator.w3.org/checklink N/A not a public link
    - [x] https://validator.w3.org N/A not a public link
    - [x] Dymanic Accessability Plugin (DAP)

### CODE REVIEW

File: src/main/content/guides.html [hunk 1]
```diff
@@ -1,29 +1,14 @@
...
 js: guides
 permalink: /guides/
 ---
 
-<!-- INTRO -->
-<div id="guides_intro_background" class="black_blue_gradient_background">
-    <div class="container">
-        <div class="row">
-            <div class="col-sm-4">
-                <img id="intro_logo" src="{{ "/img/guides_page_title_large.svg" | relative }}" class="img-responsive" alt="Guides">
```
Comment (kinueng) [inline src/main/content/guides.html]: We need to delete `guides_page_title_large.svg` if we no longer need it.

File: src/main/content/_assets/css/guides.css [hunk 2]
```diff
@@ -10,7 +10,6 @@
  *******************************************************************************/
 
 /* INTRO */
-
 #guides_intro_background {
```
Comment (kinueng) [inline src/main/content/_assets/css/guides.css]: We should delete the styling for `#guides_intro_background` and `#intro_logo`, if we no longer have the HTML.

File: src/main/content/guides.html [hunk 3]
```diff
@@ -65,8 +50,10 @@ <h3 class="guide_title">{{guide.title}}</h3>
                     <img class="duration_clock_icon" src="/img/guide_duration_clock_icon_small.svg" alt="Duration">
                     <span class="guide_duration">{{guide.duration}}</span>
                     {% if guide.layout == 'interactive-guide' %}
-                    <img class="interactive_bolt_icon" src="/img/guide_lightning_bolt.svg" alt="Interactive">
-                    <span class="guide_interactive">Interactive</span>
+                    <div class="guide_interactive_container">
+                            <img class="interactive_bolt_icon" src="/img/guide_lightning_bolt.svg" alt="Interactive">
```
Comment (kinueng) [inline src/main/content/guides.html]: Do we still need class `interactive_bolt_icon`?  I noticed it was deleted in `guide-card.css`

File: src/main/content/guides.html (hunk 1 above)
Comment (steven1046) [reply src/main/content/guides.html]: Done

File: src/main/content/_assets/css/guides.css (hunk 2 above)
Comment (steven1046) [reply src/main/content/_assets/css/guides.css]: Done

File: src/main/content/guides.html (hunk 3 above)
Comment (steven1046) [reply src/main/content/guides.html]: Yes, the only thing removed was a margin.

**Pull request merged**